cloudflare/resource_cloudflare_waiting_room_event.go

```diff
@@ -0,0 +1,205 @@
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	cloudflare "github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCloudflareWaitingRoomEvent() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudflareWaitingRoomEventCreate,
+		Read:   resourceCloudflareWaitingRoomEventRead,
+		Update: resourceCloudflareWaitingRoomEventUpdate,
+		Delete: resourceCloudflareWaitingRoomEventDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudflareWaitingRoomEventImport,
+		},
+
+		Schema: resourceCloudflareWaitingRoomEventSchema(),
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Second),
+			Update: schema.DefaultTimeout(30 * time.Second),
+		},
+	}
+}
+
+func expandWaitingRoomEvent(d *schema.ResourceData) (cloudflare.WaitingRoomEvent, error) {
+	var disableSessionRenewal *bool
+	switch b := d.Get("disable_session_renewal").(type) {
+	case bool:
+		disableSessionRenewal = &b
+	case nil:
+		disableSessionRenewal = nil
+	}
+	eventStartTime, err := time.Parse(time.RFC3339, d.Get("event_start_time").(string))
+	if err != nil {
+		return cloudflare.WaitingRoomEvent{}, err
+	}
+	eventEndTime, err := time.Parse(time.RFC3339, d.Get("event_end_time").(string))
+	if err != nil {
+		return cloudflare.WaitingRoomEvent{}, err
+	}
+	prequeueStartTime := time.Time{}
+	if t, ok := d.GetOk("prequeue_start_time"); ok {
+		prequeueStartTime, err = time.Parse(time.RFC3339, t.(string))
+		if err != nil {
+			return cloudflare.WaitingRoomEvent{}, err
+		}
+	}
+
+	return cloudflare.WaitingRoomEvent{
+		ID:                    d.Id(),
```

Comment:
i don't think this is going to always work like you intend it to. `d.Id()` is reliant on `d.SetId()` which you're calling later in the Create.

i think we can get rid of this and just set the `d.SetId()` to the ID returned from `waitingRoomEvent`.